### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by chanseokoh**
Broke down a super-long method into chunks.

Modified the file-creating methods to create folders on demand for further simplification.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java [hunk 1]
```diff
@@ -138,96 +142,83 @@ private static IFile materialize(IProject project, AppEngineProjectConfig config
...
-      for (int i = 0; i < packages.length; i++) {
-        folder = createChildFolder(packages[i], folder, subMonitor.newChild(1));
+  private static void createPomXml(IProject project, AppEngineProjectConfig config,
+      boolean isStandardProject, IProgressMonitor monitor) throws CoreException {
+    Map<String, String> properties = new HashMap<>();
+    properties.put("projectGroupId", config.getMavenGroupId()); //$NON-NLS-1$
+    properties.put("projectArtifactId", config.getMavenArtifactId()); //$NON-NLS-1$
+    properties.put("projectVersion", config.getMavenVersion()); //$NON-NLS-1$
+
+    String template;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java]: I found this a little hard to read. The template variable should at least go inside the outer if block to make the scope clearer.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java [hunk 2]
```diff
@@ -138,96 +142,83 @@ private static IFile materialize(IProject project, AppEngineProjectConfig config
...
-    monitor.subTask("Creating folder " + name);
-
-    boolean force = true;
-    boolean local = true;
-    IFolder child = parent.getFolder(name);
-    if (!child.exists()) {
-      child.create(force, local, monitor);
-    }
-    return child;
+    createChildFile("pom.xml", template, project, properties, monitor); //$NON-NLS-1$
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java]: You might instead remove the template variable and move this call into the if-else blocks.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java (hunk 1 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java]: Done.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java (hunk 2 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/CodeTemplates.java]: Done.

**Pull request merged**